WordPressKit/WordPressComServiceRemote+SiteVerticals.swift

```diff
@@ -0,0 +1,155 @@
+        self.identifier = identifier
+        self.title = title
+        self.isNew = isNew
+    }
+
+    private enum CodingKeys: String, CodingKey {
+        case identifier = "vertical_id"
+        case title      = "vertical_name"
+        case isNew      = "is_user_input_vertical"
+    }
+}
+
+// MARK: - WordPressComServiceRemote (Site Verticals)
+
+/// Describes the errors that could arise when searching for site verticals.
+///
+/// - requestEncodingFailure:   unable to encode the request parameters.
+/// - responseDecodingFailure:  unable to decode the server response.
+/// - serviceFailure:           the service returned an unexpected error.
+///
+public enum SiteVerticalsError: Error {
+    case requestEncodingFailure
+    case responseDecodingFailure
+    case serviceFailure
+}
+
+/// Advises the caller of results related to requests for site verticals.
+///
+/// - success: the site verticals request succeeded with the accompanying result.
+/// - failure: the site verticals request failed due to the accompanying error.
+///
+public enum SiteVerticalsResult {
+    case success([SiteVertical])
+    case failure(SiteVerticalsError)
+}
+
+public typealias SiteVerticalsServiceCompletion = ((SiteVerticalsResult) -> ())
+
+/// Site verticals services, exclusive to WordPress.com.
+///
+public extension WordPressComServiceRemote {
+
+    /// Retrieves Verticals matching the specified criteria.
+    ///
+    /// - Parameters:
+    ///   - request:    the value object with which to compose the request.
+    ///   - completion: a closure including the result of the request for site verticals.
+    ///
+    func retrievalVerticals(request: SiteVerticalsRequest, completion: @escaping SiteVerticalsServiceCompletion) {
+
+        let endpoint = "verticals"
+        let path = self.path(forEndpoint: endpoint, withVersion: ._2_0)
+
+        var requestParameters: [String : AnyObject]
+        do {
+            requestParameters = try encodeRequestParameters(request: request)
+        } catch {
+            DDLogError("Failed to encode \(SiteCreationRequest.self) : \(error)")
+
+            completion(.failure(SiteVerticalsError.requestEncodingFailure))
+            return
+        }
+
+        let locale = WordPressComLanguageDatabase().deviceLanguage.slug
+        requestParameters["locale"] = locale as AnyObject
+
+        wordPressComRestApi.GET(
+            path,
+            parameters: requestParameters,
+            success: { [weak self] responseObject, httpResponse in
+                DDLogInfo("\(responseObject) | \(String(describing: httpResponse))")
+
+                guard let self = self else {
+                    return
+                }
+
+                do {
+                    let response = try self.decodeResponse(responseObject: responseObject)
+                    completion(.success(response))
+                } catch {
+                    DDLogError("Failed to decode \([SiteVertical].self) : \(error.localizedDescription)")
```

Comment:
Any reason why you're logging the localizedDescription here but not in encodeRequestParameters?